amaranth/hdl/ast.py

```diff
@@ -1405,11 +1405,12 @@ class UnusedProperty(UnusedMustUse):
 class Property(Statement, MustUse):
     _MustUse__warning = UnusedProperty
 
-    def __init__(self, test, *, _check=None, _en=None, src_loc_at=0):
+    def __init__(self, test, *, _check=None, _en=None, name=None, src_loc_at=0):
         super().__init__(src_loc_at=src_loc_at)
         self.test   = Value.cast(test)
         self._check = _check
         self._en    = _en
+        self.name   = name
```

Comment:
Could you check that name is `str` or None here, please?

Reply:
Added a check, here's what it looks like when you try to use an integer:

```
Traceback (most recent call last):
  File "/home/jeanthomas/Documents/amaranth/jfng-example.py", line 7, in <module>
    m.d.sync += Assert(a != 0, name=345)
  File "/home/jeanthomas/Documents/amaranth/amaranth/hdl/ast.py", line 1415, in __init__
    raise TypeError("Property name must be a string or None, not {!r}"
TypeError: Property name must be a string or None, not 345
```